### PR TITLE
close the new page/window opened by file download action

### PR DESCRIPTION
**Pull request by wintonzheng**
> [!IMPORTANT]
> Adds logic in `handle_click_action()` to close extra pages opened during file downloads and refactors `set_download_file_listener()` for clarity.
> 
>   - **Behavior**:
>     - In `handle_click_action()` in `handler.py`, added logic to close any extra page opened during a file download action.
>     - Logs page count before and after download to identify if an extra page was opened.
>   - **Misc**:
>     - Minor refactoring in `set_download_file_listener()` in `browser_factory.py` to improve code clarity.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=Skyvern-AI%2Fskyvern&utm_source=github&utm_medium=referral)<sup> for 03b18fe3ebe2bfd004cd0b54059f916116310c30. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: skyvern/webeye/scraper/domUtils.js [hunk 1]
```diff
@@ -1134,11 +1134,7 @@ function buildElementObject(frame, element, interactable, purgeable = false) {
   }
 
   if (elementTagNameLower === "input" || elementTagNameLower === "textarea") {
-    if (element.type === "password") {
-      attrs["value"] = element.value ? "*".repeat(element.value.length) : "";
-    } else {
-      attrs["value"] = element.value;
-    }
+    attrs["value"] = element.value;
```
Comment (LawyZheng) [inline skyvern/webeye/scraper/domUtils.js:1137]: can you revert this change? it looks like something weired happened in syng PR 😂

File: skyvern/webeye/actions/handler.py [hunk 2]
```diff
@@ -358,7 +358,47 @@ async def handle_click_action(
...
+            )
+            if page_count_after_download > initial_page_count and browser_state and browser_state.browser_context:
+                LOG.info(
+                    "Extra page opened after download, closing it",
+                    task_id=task.task_id,
+                    step_id=step.step_id,
+                    workflow_run_id=task.workflow_run_id,
+                )
+                # close the extra page
+                await browser_state.browser_context.pages[-1].close()
```
[the remark this answers is not in the document]
Comment (wintonzheng) [reply skyvern/webeye/actions/handler.py:401]: i'll add a warning if it's not
